### PR TITLE
Login input cleaning fix

### PR DESCRIPTION
**Pull request by EmmanuelWLX**
## Summary
- Fixed the cleaning of inputs from login screen. Now it's almost imperceptible, I think

### Android
loginInputClean-nexus6p

### iOS
loginInputClean-iphone6

### CODE REVIEW

File: LibraryApp/src/app/screens/Login/index.js [hunk 1]
```diff
@@ -3,50 +3,41 @@ import PropTypes from 'prop-types';
...
 class LoginContainer extends Component {
   state = {
-    email: '',
-    password: '',
     emailError: null,
     passwordError: null
   };
 
-  onEmailChange = email => this.setState({ email });
+  onEmailChange = email => this.props.setEmail(email);
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/Login/index.js]: if you were using redux form this wont be needed

File: LibraryApp/src/redux/auth/reducer.js [hunk 2]
```diff
@@ -5,15 +5,28 @@ import { actions } from './actions';
...
 
 const reducer = (state = immutable(initialState), action) => {
+  const { payload } = action;
   switch (action.type) {
-    case actions.INIT_STORED_USER:
+    case actions.AUTH_INIT:
+      return state.merge({
+        initialAuthLoading: true
+      });
+    case actions.AUTH_INIT_SUCCESS:
```
Comment (mmamaniwolox) [inline LibraryApp/src/redux/auth/reducer.js]: You should use redux recompose here to avoid this

Comment (EmmanuelWLX) [reply LibraryApp/src/redux/auth/reducer.js]: Should I do it even if i don't need initialAuth or initialAuthError variables?

File: LibraryApp/src/app/components/CustomTextInput/index.js [hunk 3]
```diff
@@ -6,13 +6,14 @@ import styles from './styles';
 
 class CustomTextInput extends Component {
   state = {
-    value: this.props.value
+    value: this.props.value || (this.props.input && this.props.input.value)
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/components/CustomTextInput/index.js]: Use `getDerivedStateFromProps` instead

File: LibraryApp/src/redux/auth/reducer.js [hunk 4]
```diff
@@ -1,41 +1,33 @@
...
+  primaryActions: [actions.LOGIN, actions.AUTH_INIT],
+  override: {
+    [actions.AUTH_INIT_SUCCESS]: (state, action) =>
+      state.merge({
+        initialAuth: action.payload,
+        initialAuthLoading: false,
+        initialAuthError: null,
+        loginLoading: false
+      }),
+    [actions.LOGIN_SUCCESS]: (state, action) =>
```
Comment (mmamaniwolox) [inline LibraryApp/src/redux/auth/reducer.js]: I dont think you need this

File: LibraryApp/src/redux/auth/reducer.js [hunk 5]
```diff
@@ -1,41 +1,33 @@
...
-    case actions.LOGOUT:
-      return state.merge({ ...initialState });
-    default:
-      return state;
+const initialState = completeState(stateDescription);
+
+const reducerDescription = {
+  primaryActions: [actions.LOGIN, actions.AUTH_INIT],
+  override: {
+    [actions.AUTH_INIT_SUCCESS]: (state, action) =>
```
Comment (mmamaniwolox) [inline LibraryApp/src/redux/auth/reducer.js]: dont need this either

File: LibraryApp/src/redux/auth/reducer.js (hunk 4 above)
Comment (EmmanuelWLX) [reply LibraryApp/src/redux/auth/reducer.js]: you was right! i forgot the target attr in the onSuccess() override 😓

**Pull request merged**